lib/simplecov.rb

```diff
@@ -58,6 +58,38 @@ def start(profile = nil, &block)
       end
     end
 
+    #
+    # Collate a series of SimpleCov result files into a single SimpleCov output.
+    # You can optionally specify configuration with a block:
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"]
+    #    OR
+    #   SimpleCov.start Dir["simplecov-resultset-*/.resultset.json"], 'rails' # using rails profile
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"] do
+    #     add_filter 'test'
+    #   end
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"], 'rails' do
+    #     add_filter 'test'
+    #   end
+    #
+    # Please check out the RDoc for SimpleCov::Configuration to find about available config options
+    #
+    def collate(result_filenames, profile = nil, &block)
+      load_profile(profile) if profile
+      configure(&block) if block_given?
+      self.running = true
+      self.pid = Process.pid
+
+      results = result_filenames.map { |filename| JSON.parse(File.read(filename)) || {} }.reduce(&:merge).map do |command_name, data|
+        # Re-create each included instance of SimpleCov::Result from the stored run data.
+        SimpleCov::Result.from_hash(command_name => data)
+      end
+
+      # Use the ResultMerger to produce a single, merged result, ready to use.
+      @result = SimpleCov::ResultMerger.merge_results(*results)
```

Comment:
me typing...

```ruby
  #   SimpleCov.start 'app'
  #   STDERR.puts '[COVERAGE] Running'
  #   configure_to_only_generate_result!
  def configure_to_only_generate_result!
   SimpleCov.configure do
      # use_merging   true
      minimum_coverage 0.0 # disable
      maximum_coverage_drop 100.0 # disable
      Instance_eval(Proc.new) if block_given?
    end
    SimpleCov.formatters = []
    SimpleCov.at_exit do
      STDERR.puts "[COVERAGE] creating #{File.join(SimpleCov.coverage_dir, '.resultset.json')}"
      SimpleCov.result.format!
    end
  end

  # generate_report!(result_filename) do
  #    minimum_coverage ENV.fetch('COVERAGE_MINIMUM') { 100.0 }.to_f.round(2)
  #    minimum_coverage_by_file 60
  #    maximum_coverage_drop 1
  #    refuse_coverage_drop
  #    SimpleCov.formatters = [SimpleCov::Formatter::HTMLFormatter]
  # end
  def generate_report!(result_filenames)
      Proc.new.call if block_given?
      merged_result = merge_results_files(result_filenames)
      if SimpleCov.formatters.any?
        merged_result.format!
      end
      process_result(merged_result)
  end

  def merge_results_files(results_files)
    results = results_files.map {|results_file|
      next unless File.exist?(results_file)
      json = JSON.parse(File.read(results_file))
      STDERR.puts "[COVERAGE] merging #{results_file}; processing..."
      SimpleCov::Result.from_hash(json)
    }.compact
    if results.any?
      SimpleCov::ResultMerger.merge_results(*results)
    else
      abort "No files found to report: #{results_files.inspect}"
    end
  end

  def process_result(result)
    @exit_status = SimpleCov.process_result(result, SimpleCov::ExitCodes::SUCCESS)

    # Force exit with stored status (see github issue #5)
    # unless it's nil or 0 (see github issue #281)
    Kernel.exit @exit_status if @exit_status && @exit_status > 0
  end
```

Reply:
@bf4 sorry I'm not sure what this is supposed to tell me/us? :thinking: Is there something missing from this PR?